wallet/src/libwallet/internal/updater.rs

```diff
@@ -94,19 +94,12 @@ where
 {
 	// just read the wallet here, no need for a write lock
 	let mut txs = if let Some(id) = tx_id {
-		let tx = wallet.tx_log_iter().find(|t| t.id == id);
-		if let Some(t) = tx {
-			vec![t]
-		} else {
-			vec![]
-		}
+		wallet.tx_log_iter().filter(|t| t.id == id).collect()
```

Comment:
Oh... there's 2 txs in the tx log here for a self send right?
:+1:

Reply:
Yes. The receive tx ends up with a lower ID in the self-send via http case, and previous short circuit find was just returning the first matching TX it found, which is a recieve tx. The finalize is explicitly looking for a send tx.